cms/src/main/java/uk/nhs/digital/ps/modules/SearchableFlagModule.java

```diff
@@ -0,0 +1,66 @@
+package uk.nhs.digital.ps.modules;
+
+import org.onehippo.cms7.services.HippoServiceRegistry;
+import org.onehippo.cms7.services.eventbus.HippoEventBus;
+import org.onehippo.cms7.services.eventbus.Subscribe;
+import org.onehippo.repository.events.HippoWorkflowEvent;
+import org.onehippo.repository.modules.DaemonModule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.nhs.digital.ps.modules.searchableFlagModule.DatasetProcessor;
+import uk.nhs.digital.ps.modules.searchableFlagModule.PublicationProcessor;
+
+import java.util.Arrays;
+import java.util.List;
+import javax.jcr.*;
+
+public class SearchableFlagModule implements DaemonModule {
+
+    private static final Logger log = LoggerFactory.getLogger(SearchableFlagModule.class);
+
+    private static final String ACTION_COMMIT = "commitEditableInstance";
+    private static final String ACTION_PUBLISH = "publish";
+    private static final String STATE_PUBLISHED = "published";
+    private static final String STATE_DRAFT = "draft";
+    private static final String STATE_UNPUBLISHED = "unpublished";
+
+    private Session session;
+    private final PublicationProcessor publicationUpdater = new PublicationProcessor();
+    private final DatasetProcessor datasetUpdater = new DatasetProcessor();
+
+    @Override
+    public void initialize(final Session session) throws RepositoryException {
+        this.session = session;
+        HippoServiceRegistry.registerService(this, HippoEventBus.class);
+    }
+
+    @Override
+    public void shutdown() {
+        HippoServiceRegistry.unregisterService(this, HippoEventBus.class);
+    }
+
+    @Subscribe
+    public void handleEvent(final HippoWorkflowEvent event) {
+        if (event.success() && ACTION_COMMIT.equals(event.action())) {
+            updateSearchableFlag(event, Arrays.asList(STATE_UNPUBLISHED, STATE_DRAFT));
```

Comment:
As acceptedStates is always STATE_UNPUBLISHED, STATE_DRAFT and doesn't change, you could have this as a static that you check in 
`hasValidState(Node node, List<String> validStates)`
and not bother passing it through several methods as a parameter? Might be easier to read that way.